lib/dispatcher/dispatcher.js

```diff
@@ -14,6 +13,32 @@ class Dispatcher extends EventEmitter {
   destroy () {
     throw new Error('not implemented')
   }
+
+  compose (...args) {
+    // So we handle [interceptor1, interceptor2] or interceptor1, interceptor2, ...
+    const interceptors = Array.isArray(args[0]) ? args[0] : args
+    let dispatch = this.dispatch.bind(this)
+
+    for (const interceptor of interceptors) {
+      if (interceptor == null) {
+        continue
+      }
+
+      if (typeof interceptor !== 'function') {
```

Comment:
I still think we should allow return dispatcher shape?

Reply:
Hmm, not sure, but maybe I'm not seeing the full picture; what would be the benefit of exposing the whole dispatcher shape?

Reply:
Not sure, mostly if the interceptor needs some kind of disposing.

Reply:
How it should work?
Meaning, if I return a `dispatcher`-like, how the dispatcher should handle the rest of the properties, given that it will just allow to _patch_ the `dispatch` function 🤔